File: riak/perl/read/riak_read.pl
#! /usr/bin/perl
#
#	read/riak_read.pl
#
#					Mar/14/2013
# -------------------------------------------------------
use	strict;
use	JSON;
use	warnings;
use	utf8;
use	Encode;
#
use lib '/var/www/data_base/common/perl_common';
use uri_get;
#
#
# -------------------------------------------------------
print	(encode ('utf-8',"*** 開始 ***\n"));
#
my $url_base = 'http://localhost:8098/riak/shimane';
my $url_in = $url_base . '?keys=true';
#
my $str_json = uri_get::uri_get_proc ($url_in);
#
my %data_aa = %{decode_json ($str_json)};
#
my @keys = @{$data_aa{"keys"}};
my $num = $#keys;
print "num = " . $num . "\n";
#
foreach my $tkey (@keys){
	my $url_city = $url_base . '/' . $tkey ;
	my $str_json_city = uri_get::uri_get_proc ($url_city);
#	print $str_json_city . "\n";
	my %data_city = %{decode_json ($str_json_city)};
	my $out_str = $tkey . "\t";
	$out_str .=  $data_city{"name"} . "\t";
	$out_str .=  $data_city{"population"} . "\t";
	$out_str .=  $data_city{"date_mod"} . "\n";
	print (encode ('utf-8',$out_str));
	}
#
print	(encode ('utf-8',"*** 終了 ***\n"));
#
# -------------------------------------------------------

File: riak/java/read/riak_read.java
// --------------------------------------------------------------
//	read/riak_read.java
//
//					Feb/24/2015
// --------------------------------------------------------------
import	java.util.HashMap;
import	java.util.ArrayList;
import  net.arnx.jsonic.JSON;
// --------------------------------------------------------------
public class riak_read
{

// --------------------------------------------------------------
public static void main (String[] args)
{
	System.out.println ("*** 開始 ***");

	final String url_base = "http://host_ubuntu1:8098/riak/shimane";
	final String url_in = url_base + "?keys=true";

	String str_json = get_uri.get_uri_proc (url_in);
//	System.out.println (str_json);

/*
try
	{
	HashMap <String, Object> 
	data_aa = (HashMap <String, Object>)JSON.decode (str_json);
//	System.out.println (data_aa.get ("keys"));
	ArrayList keys = (ArrayList) data_aa.get ("keys");

	int nn = keys.size ();

	System.out.println (nn);
	for (int it=0; it< nn; it++)
		{
		String key = (String)keys.get (it);
		System.out.print (key + "\t");
		final String url_city = url_base + "/" + key;
		String str_json_city = get_uri.get_uri_proc (url_city);
		HashMap <String, Object> data_city
		= (HashMap <String, Object>)JSON.decode (str_json_city);
		System.out.print (data_city.get ("name") + "\t" );
		System.out.print (data_city.get ("population") + "\t");
		System.out.println (data_city.get ("date_mod") );
		}
	}
	catch (Exception ee)
	{
	ee.printStackTrace ();
	}
*/

	System.out.println ("*** 終了 ***");
}

// --------------------------------------------------------------
}
// --------------------------------------------------------------
